frontend/src/pages/WaitingPage.tsx

```diff
@@ -4,12 +4,15 @@ import { CircularProgress, Typography } from "@mui/material";
 import { Stack } from "@mui/system";
 
 import { useSocket } from "src/contexts/WsContext";
+import { useGetUsername } from "src/hooks/useUsers";
 
 import { Match } from "~shared/types/api/match.dto";
 
 export const WaitingPage = () => {
   const navigate = useNavigate();
   const { socket, connect } = useSocket();
+  const { getUsername } = useGetUsername();
```

Comment:
If we use `react-query`, then we will need to `useState` to fire off the queries when the user IDs become available.

```suggestion
  const [userOne, setUserOne] = useState<number | undefined>(undefined);
  const [userTwo, setUserTwo] = useState<number | undefined>(undefined);
  const [roomId, setRoomId] = useState<string | undefined>(undefined);
  const { user: userNameOne } = useGetUserName(userOne);
  const { user: userNameTwo } = useGetUserName(userTwo);
```